scipy/optimize/_qap.py

```diff
@@ -0,0 +1,408 @@
+                sorting of input adjacency matrices, gives users the option
+                to shuffle the nodes of `cost_matrix`. Results are then
+                unshuffled so that returned `col_ind` matches the node order
+                of inputs.
+            eps : float (default = 0.05)
+                A positive, threshold stopping criteria such that Franke-
+                Wolfe continues to iterate while Frobenius norm of
+                :math:`(P_{i}-P_{i+1}) > eps`, where :math:`i` is the
+                iteration number.
+
+    Returns
+    -------
+    res : OptimizeResult
+        A :class:`scipy.optimize.OptimizeResult` consisting of the fields:
+
+            col_ind : 1-D array
+                An array of column indices corresponding to the optimal
+                permutation (with the fixed seeds given) of the
+                nodes of `dist_matrix`, to best minimize the objective
+                function.
+            score : float
+                The optimal value of the objective function.
+            nit : int
+                The total number of Franke-Wolfe iterations performed during
+                optimization.
+
+    References
+    ----------
+    .. [1] J.T. Vogelstein, J.M. Conroy, V. Lyzinski, L.J. Podrazik,
+           S.G. Kratzer, E.T. Harley, D.E. Fishkind, R.J. Vogelstein, and
+           C.E. Priebe, "Fast approximate quadratic programming for graph
+           matching," PLOS one, vol. 10, no. 4, p. e0121002, 2015,
+           https://doi.org/10.1371/journal.pone.0121002
+
+    .. [2] D. Fishkind, S. Adali, H. Patsolic, L. Meng, D. Singh, V. Lyzinski,
+           C. Priebe, "Seeded graph matching", Pattern Recognit. 87 (2019):
+           203-215, https://doi.org/10.1016/j.patcog.2018.09.014
+
+    Examples
+    --------
+
+    >>> cost = np.array([[0, 80, 150, 170], [80, 0, 130, 100],
+    ...         [150, 130, 0, 120], [170, 100, 120, 0]])
+    >>> dist = np.array([[0, 5, 2, 7], [0, 0, 3, 8],
+    ...         [0, 0, 0, 3], [0, 0, 0, 0]])
+    >>> from scipy.optimize import quadratic_assignment
+    >>> res = quadratic_assignment(cost, dist)
+    >>> print(res)
+     col_ind: array([0, 3, 2, 1])
+         nit: 9
+       score: 3260
+
+    To demonstrate explicitly how the `score` value
+    :math:`f(P) = trace(A^T PBP^T )` is calculated, one may construct the
+    permutation matrix, and perform the necessary algebra.
+
+    >>> n = cost.shape[0]
+    >>> P = np.zeros((n, n))
+    >>> P[np.arange(n), res['col_ind']] = 1
+    >>> score = int(np.trace(cost.T @ P @ dist @ P.T))
+    >>> print(score)
+    3260
+
+    As you can see, the value here matches res['score'] reported above.
+    Alternatively, to avoid constructing the permutation matrix, one can also
+    perform the following calculation.
+
+    >>> score = np.trace(cost.T @ dist[np.ix_(res['col_ind'], res['col_ind'])])
+    >>> print(score)
+    3260
+
+    Here, we are simply permuting the distance matrix.
+
```

Comment:
Maybe it would be nice to show that this happens to be the optimum for this small problem.

```
perm_opt, score_opt = None, np.inf
for perm in permutations([0, 1, 2, 3]):
    P = np.eye(n)[:, perm]
    score = int(np.trace(cost @ P @ dist.T @ P.T))
    if score < score_opt:
        score_opt, perm_opt = score, perm
print( perm_opt == res['col_ind'].tolist())
```

Reply:
I'm going to submit a PR to your repo. I'll include this and other things there.